polaris/benchmark/_base.py

```diff
@@ -180,28 +181,33 @@ def _validate_split(cls, v, info: ValidationInfo):
         ):
             raise InvalidBenchmarkError("The predefined split contains empty test partitions")
 
-        if len(v[0]) == 0:
+        train_idx_list = v[0]
+        test_idx_list = list(i for part in v[1].values() for i in part) if isinstance(v[1], dict) else v[1]
+
+        if len(train_idx_list) == 0:
             logger.info(
                 "This benchmark only specifies a test set. It will return an empty train set in `get_train_test_split()`"
             )
 
-        train_indices = v[0]
-        test_indices = [i for part in v[1].values() for i in part] if isinstance(v[1], dict) else v[1]
+        train_idx_set = set(train_idx_list)
+        test_idx_set = set(test_idx_list)
 
         # The train and test indices do not overlap
-        if any(i in train_indices for i in test_indices):
+        if len(train_idx_set & test_idx_set) > 0:
             raise InvalidBenchmarkError("The predefined split specifies overlapping train and test sets")
 
         # Duplicate indices
-        if len(set(train_indices)) != len(train_indices):
+        if len(train_idx_set) != len(train_idx_list):
             raise InvalidBenchmarkError("The training set contains duplicate indices")
-        if len(set(test_indices)) != len(test_indices):
+        if len(test_idx_set) != len(test_idx_list):
             raise InvalidBenchmarkError("The test set contains duplicate indices")
 
         # All indices are valid given the dataset
         if info.data["dataset"] is not None:
-            if any(i < 0 or i >= len(info.data["dataset"]) for i in train_indices + test_indices):
+            max = len(info.data["dataset"])
```

Comment:
**Suggestion:** I would rename this variable.

```suggestion
            max_i = len(info.data["dataset"])
```


Naming a variable `max` overwrites the built-in `max()` function. E.g.:

```
>>> max(1, 3)
3
>>> max = 3
>>> max(1, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not callable
```

Reply:
Good catch!